### PR TITLE
Add Add Tag

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add a floating action button to the **_Edit Tags_** screen to add a tag.  See the animation below for illustration.

Fast|Slow
-|-
<kbd><a href="https://user-images.githubusercontent.com/3827611/100007866-ef178e80-2d89-11eb-9d2a-b6c10a77e814.gif"></a></kbd>|<kbd><a href="https://user-images.githubusercontent.com/3827611/100007851-e6bf5380-2d89-11eb-9189-95a358c3caa0.gif"></a></kbd>

When a valid tag name is input and the **_Save_** button is tapped in the **_Add Tag_** dialog, the added tag is shown in the list on the **_Edit Tags_** screen.  See the screenshots below for illustration.

add_add_tag

The **_Add Tag_** dialog will show an error message and disable the **_Save_** button if the tag is empty, contains spaces, already exists, or is too long.  See the screenshots below for illustration.

add_add_tag_error

### Test
1. Open navigation drawer.
2. Tap ***Edit*** button in ***Tags*** header.
3. Notice floating action button in bottom, right/end corner or ***Edit Tags*** screen.
4. Long-press floating action button.
5. Notice ***Add Tag*** message is shown.
6. Tap floating action button.
7. Notice ***Add Tag*** dialog is shown.
8. Enter space character in ***Tag name*** field.
9. Notice ***Tags cannot contain spaces*** error message is shown.
10. Notice ***Save*** button is disabled.
11. Delete space character from ***Tag name*** field.
12. Notice ***Tags cannot be empty*** error message is shown.
13. Notice ***Save*** button is disabled.
14. Enter any character in ***Tag name*** field.
15. Notice ***Save*** button is enabled.
16. Enter existing tag name in ***Tag name*** field.
17. Notice ***Tag already exists*** error message is shown.
18. Notice ***Save*** button is disabled.
19. Enter four England flag emojis (i.e. `🏴󠁧󠁢󠁥󠁮󠁧󠁿🏴󠁧󠁢󠁥󠁮󠁧󠁿🏴󠁧󠁢󠁥󠁮󠁧󠁿🏴󠁧󠁢󠁥󠁮󠁧󠁿`) in ***Tag name*** field.
20. Notice ***Tag length too long*** error message is shown.
21. Notice ***Save*** button is disabled.
22. Enter non-existing tag name in ***Tag name*** field.
23. Notice ***Save*** button is enabled.
24. Tap ***Save*** button.
25. Notice tag from Step  is added to list.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

#### Note
Contact me for an installable build.

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import android.widget.ImageButton;
+
+import androidx.annotation.NonNull;
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
+
+import com.google.android.material.snackbar.Snackbar;
+
+import java.util.List;
+
+public class ImageButtonBehavior extends CoordinatorLayout.Behavior<ImageButton> {
```
Comment (hichamboushaba) [inline Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java]: Can you elaborate on the use of this class?
I understand from its code that it tries to move the "add" button if a snackbar is displayed, but from what I see, it doesn't do anything in the current state as the ImageButton it's applied to is not a child of a CoordinatorLayout.
And a question, do we actually need it? as I didn't find any way to trigger a SnackBar when viewing TagsActivity.

File: Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        ThemeUtils.setTheme(this);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_tag_add);
+
+        int widthScreen = getResources().getDisplayMetrics().widthPixels;
+        int widthMargin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 56, getResources().getDisplayMetrics());
+        int widthMaximum = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 312, getResources().getDisplayMetrics());
+        LinearLayout layout = findViewById(R.id.layout);
+        layout.getLayoutParams().width = Math.min(widthMaximum, widthScreen - widthMargin);
+        layout.requestLayout();
```
Comment (hichamboushaba) [inline Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java:51]: just a np, why ConstraintLayouts are not used in the project? if they were used, you can have this logic in the XML layout, and would be easier to understand.

**Review by hichamboushaba — COMMENTED** [review]
Nice work, the animation looks great 👍 
I left two comments, please check them before approving 🙂

**Review by SylvesterWilmott — APPROVED** [review]
Design looks good! Thanks.

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java (hunk 1 above)
Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java]: You're right.  This isn't utilized at the moment.  It was included to future-proof the case when a snackbar was used on the ***Edit Tags*** screen like for an ***Undo*** action when deleting a tag.

File: Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java (hunk 2 above)
Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java:51]: `ConstraintLayout`s were never required to do what we need to with Simplenote.  So, they weren't added.  It's debatable which view layout is more understandable when you've been working on Android before the time of `ConstraintLayout` and you're used to `LinearLayout` and `RelativeLayout` for everything.

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java (hunk 1 above)
Comment (hichamboushaba) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java]: IMO it's better to not add it to the code until the undo feature is added, as the changes are more than just the behavior: CoordinatorLayout should be used too.
And the least is to add a comment explaining that it's not used now.

Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java]: I removed the `ImageButtonBehavior` class in ef90cdd.

Comment (hichamboushaba) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java]: Sorry for nitpicking, but the ImageButton has still the app:layout_behavior property :).

I'll approve the PR, you can merge it when it's removed

**Review by hichamboushaba — APPROVED** [review]

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java (hunk 1 above)
Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/ImageButtonBehavior.java]: No need to apologize.  That's why code reviews exist.  We shouldn't leave that attribute in the code if the class doesn't exist.  I removed the attribute in 5b4a969.

**Pull request merged**